test_scripts/API/ServiceStatusUpdateToHMI/common.lua

```diff
@@ -0,0 +1,244 @@
+          test:initHMI_onReady()
+          :Do(function()
+              utils.cprint(35, "HMI is ready")
+              m.getHMIConnection():SendNotification("BasicCommunication.OnSystemTimeReady")
+              test:connectMobile()
+              :Do(function()
+                  utils.cprint(35, "Mobile connected")
+                  m.allowSDL()
+                end)
+            end)
+        end)
+    end)
+end
+
+function m.ptUpdate(pTbl)
+  local filePath = "./files/Security/client_credential.pem"
+  local crt = utils.readFile(filePath)
+  pTbl.policy_table.module_config.certificate = crt
+end
+
+local preconditionsOrig = common.preconditions
+function m.preconditions(pForceProtectedServices)
+  preconditionsOrig()
+  if not pForceProtectedServices then pForceProtectedServices = "Non" end
+  m.setSDLIniParameter("ForceProtectedService", pForceProtectedServices)
+end
+
+local postconditionsOrig = common.postconditions
+function m.postconditions()
+  postconditionsOrig()
+  m.restoreSDLIniParameters()
+end
+
+function m.decryptCertificateRes(pId, pMethod)
+  m.getHMIConnection():SendResponse(pId, pMethod, "SUCCESS", { })
+end
+
+function m.policyTableUpdateSuccess(pPTUpdateFunc)
+  local function expNotificationFunc()
+    m.getHMIConnection():ExpectRequest("BasicCommunication.DecryptCertificate")
+    :Do(function(_, data)
+        m.decryptCertificateRes(data.id, data.method)
+      end)
+    :Times(AtMost(1))
+    m.getHMIConnection():ExpectRequest("VehicleInfo.GetVehicleData", { odometer = true })
+  end
+  m.getHMIConnection():ExpectRequest("BasicCommunication.PolicyUpdate")
+  :Do(function(e, data)
+      if e.occurences == 1 then
+        m.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { })
+        m.policyTableUpdate(pPTUpdateFunc, expNotificationFunc)
+      end
+    end)
+end
+
+function m.policyTableUpdateUnsuccess()
+  local pPTUpdateFunc = function(pTbl)
+    pTbl.policy_table.app_policies = nil
+  end
+  local expNotificationFunc = function()
+    common.getHMIConnection():ExpectRequest("VehicleInfo.GetVehicleData")
+    :Times(0)
+    common.getHMIConnection():ExpectRequest("BasicCommunication.DecryptCertificate")
+    :Times(0)
+  end
+  common.getHMIConnection():ExpectRequest("BasicCommunication.PolicyUpdate")
+  :Do(function(e, data)
+      if e.occurences == 1 then
+        common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { })
+        common.policyTableUpdate(pPTUpdateFunc, expNotificationFunc)
+      end
+    end)
+  :Times(AtLeast(1))
+end
+
+function m.policyTableUpdateFunc()
+  m.getHMIConnection():ExpectNotification("SDL.OnStatusUpdate",
+    { status = "UPDATE_NEEDED" }, { status = "UPDATING" }, { status = "UP_TO_DATE" })
+  :Times(3)
+
+  m.policyTableUpdateSuccess(m.ptUpdate)
+end
+
+function m.startServiceFunc(pServiceId, pAppId)
+  m.getMobileSession(pAppId):StartSecureService(pServiceId)
+end
+
+m.serviceData = {
+  [7] = {
+    forceCode = "Non",
```

Comment:
@jacobkeeler, `forceCode` contains value for `ForceProtectedService` and is used in `precondition`, RPC service can't be force protected because RPC service should be started as not protected and only afterwards it can be set to protected.
So `forceCode=Non` means that `ForceProtectedService=Non` is set from the SDL start.